crates/transaction-generator-lib/src/args.rs

```diff
@@ -43,23 +44,32 @@ impl TransactionTypeArg {
             },
             TransactionTypeArg::CustomFunctionLargeModuleWorkingSet => {
                 TransactionType::CallCustomModules {
+                    initial_entry_point: None,
                     entry_point: EntryPoints::Nop,
                     num_modules: 1000,
                     use_account_pool: false,
                 }
             },
             TransactionTypeArg::CreateNewResource => TransactionType::CallCustomModules {
+                initial_entry_point: None,
                 entry_point: EntryPoints::BytesMakeOrChange {
                     data_length: Some(32),
                 },
                 num_modules: 1,
                 use_account_pool: true,
             },
             TransactionTypeArg::NoOp => TransactionType::CallCustomModules {
+                initial_entry_point: None,
                 entry_point: EntryPoints::Nop,
                 num_modules: 1,
                 use_account_pool: false,
             },
+            TransactionTypeArg::TokenV1NFTMint => TransactionType::CallCustomModules {
+                initial_entry_point: Some(EntryPoints::InitializeCollection),
```

Comment:
initial_entry_point is really uniquely defined based on an entry_point itself.

So, instead of having initial_entry_point here that we would need to specify, I would make entry_point know what needs to be initialized - i.e. have a function that corresponds to use_specific_transaction, called initialize_specific_transaction, that returns Option, based on whether entry_point needs any initialization or not (or entry_point to have get_initialize_entry_point to return Option<EntryPoint>, though I think the first option is a bit cleaner)

Reply:
Done